spec/lib/common/subscription-spec.js

```diff
@@ -2,20 +2,35 @@
 
 
 'use strict';
-
 describe('Subscription', function () {
     var Subscription;
-
-    helper.before();
+    var Logger;
+    var sandbox = sinon.sandbox.create();
+
+    helper.before(function (context) {
+        context.MessengerServices = function() {
+            this.start= sandbox.stub().resolves();
+            this.stop = sandbox.stub().resolves();
+            this.publish = sandbox.stub().resolves();
+        };
+        return [
+            helper.di.simpleWrapper(context.MessengerServices, 'Messenger')
+        ];
+    });
 
     before(function () {
         Subscription = helper.injector.get('Subscription');
 
-        this.options = {
+        Logger = helper.injector.get('Logger');
+        Logger.prototype.warning = sinon.stub().returns(Promise.resolve());
```

Comment:
I don't know if it's really in use or would be helpful here, but we do have a mock logger at https://github.com/RackHD/on-core/blob/master/spec/mocks/logger.js. Anyways, it's often easier to just mock out `Logger.prototype.log` than the individual log level methods (since they just call down into log).